one_fm/overrides/leave_application.py

```diff
@@ -431,7 +431,8 @@ def on_cancel(self):
             reassign_responsiobility.reassign()
         self.create_leave_ledger_entry(submit=False)
         # notify leave applier about cancellation
-        leave_application_on_cancel()
+        method = "on_cancel"
```

Comment:
why do you need to declare this particular variable here

Reply:
The leave_application_on_cancel() function was previously invoked via a hook, where we performed the Hajj check for the leave application. This check happens in two cases:

When cancelling the leave.
When submitting the leave.
The hajj_check() function takes two parameters: doc (the leave application document) and method (the action being performed, such as "submit" or "cancel"). To differentiate between the cancellation and submission scenarios, we explicitly pass the method (cancel or submit) to the hajj_check() function.

Here’s how we ensure that the Hajj check is executed correctly for both actions:

When cancelling a leave, we set the method parameter to "on_cancel".
When submitting a leave, we set the method parameter to "on_submit".
This way, the hajj_check() function knows which action is being performed, allowing it to run the appropriate logic for each case.

Reply:
@samdanikouser , if that be the case, for this, eave_application_on_cancel(self,method)


you can just do this, eave_application_on_cancel(self,"on cancel"), instead of declaring another variable, or is it being used in multiple places

Reply:
Notes. Changes done !